doc/src/tutorial/codegen_explainatory.rst

```diff
@@ -0,0 +1,458 @@
+directory, and leave the files intact when finished. For instance::
+
+    >>> from sympy.utilities.autowrap import autowrap
+    >>> from sympy.physics.qho_1d import psi_n
+    >>> from sympy import IndexedBase, Idx
+    >>> from sympy import Eq
+    >>> from sympy import symbols
+    >>> x=IndexedBase('x')
+    >>> y=IndexedBase('y')
+    >>> m=symbols('m', integer=True)
+    >>> i = Idx('i', m)
+    >>> a,omega= symbols('a, omega')
+    >>> qho=autowrap(Eq(y[i], psi_n(0, x[i], m, omega)), tempdir='/home/thd/Documents/tmp')
+
+Checking the Fortran source code in the directory specified reveals this::
+
+    subroutine autofunc(m, omega, x, y)
+    implicit none
+    INTEGER*4, intent(in) :: m
+    REAL*8, intent(in) :: omega
+    REAL*8, intent(in), dimension(1:m) :: x
+    REAL*8, intent(out), dimension(1:m) :: y
+    INTEGER*4 :: i
+
+    REAL*8, parameter :: hbar = 1.05457162d-34
+    REAL*8, parameter :: pi = 3.14159265358979d0
+    do i = 1, m
+       y(i) = (m*omega)**(1.0d0/4.0d0)*exp(-4.74126166983329d+33*m*omega*x(i &
+             )**2)/(hbar**(1.0d0/4.0d0)*pi**(1.0d0/4.0d0))
+    end do
+
+    end subroutine
+
+Using the argument 'args' along with it changes argument sequence::
+
+    >>> qho=autowrap(Eq(y[i], psi_n(0, x[i], m, omega)), tempdir='/home/thd/Documents/tmp',args=[y,x,m,omega])
+
+yields::
+
+    subroutine autofunc(y, x, m, omega)
+    implicit none
+    INTEGER*4, intent(in) :: m
+    REAL*8, intent(in) :: omega
+    REAL*8, intent(out), dimension(1:m) :: y
+    REAL*8, intent(in), dimension(1:m) :: x
+    INTEGER*4 :: i
+
+    REAL*8, parameter :: hbar = 1.05457162d-34
+    REAL*8, parameter :: pi = 3.14159265358979d0
+    do i = 1, m
+       y(i) = (m*omega)**(1.0d0/4.0d0)*exp(-4.74126166983329d+33*m*omega*x(i &
+             )**2)/(hbar**(1.0d0/4.0d0)*pi**(1.0d0/4.0d0))
+    end do
+
+    end subroutine
+
+The argument ``verbose`` is boolean, optional and if True, autowrap 
+will not mute the command line backends. This can be helpful for debugging.
+
+The argument ``language`` and ``backend`` are used to change defaults: 'Fortran'
+and 'f2py' to 'C' and 'Cython'.
+The argument helpers is used to define auxillary expressions needed for the main 
+expression. If the main expression needs to call a specialized function it should 
+be put in the ``helpers`` iterable. Autowrap will then make sure that the
+compiled main expression can link to the helper routine. Items should
+be tuples with (<funtion_name>, <sympy_expression>, <arguments>). It is mandatory 
+to supply an argument sequence to helper routines.
+
+Another method available at the ``autowrap`` level is ``binary_function``. It returns 
+a sympy function. The advantage is that we can have very fast functions as compared
+to SymPy speeds. An illustration::
+
+    >>> from sympy.utilities.autowrap import binary_function
+    >>> from sympy import symbols
+    >>> from sympy.physics.hydrogen import R_nl
+    >>> a,r = symbols('a,r')
+    >>> psi_nl=R_nl(1,0,a,r)
```

Comment:
Please go through your code snippets and make them PEP8 compatible, for example this line should be:

`>>> psi_n1 = R_n1(1, 0, a, r)`